app/test_main.py

```diff
@@ -2,0 +1,77 @@
+from typing import Callable
+
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.fixture()
+def mocked_valid_url() -> None:
+    with mock.patch("app.main.valid_google_url") as mock_valid_url:
+        yield mock_valid_url
+
+
+@pytest.fixture()
+def mocked_has_connection() -> None:
+    with mock.patch("app.main.has_internet_connection") as mock_has_connection:
+        yield mock_has_connection
+
+
+@pytest.mark.parametrize(
+    "validation,connection,url,result",
+    [
+        pytest.param(
+            False,
+            False,
+            "https://www.google.com.ua",
+            "Not accessible",
+            id="Should be NOT ACCESSIBLE when no connection and url not valid"
+        ),
+        pytest.param(
+            True,
+            False,
+            "https://www.google.com.ua",
+            "Not accessible",
+            id="Should be NOT ACCESSIBLE when url not valid"
```

Comment:
The test case id `Should be NOT ACCESSIBLE when url not valid` is not accurate. The URL is valid in this case, but the internet connection is not available. Consider updating the id to `Should be NOT ACCESSIBLE when connection is not available and url is valid`.